jdaviz/core/loaders/importers/image/image.py

```diff
@@ -0,0 +1,230 @@
+
+    # HDUList-specific options
+    input_hdulist = Bool(False).tag(sync=True)
+    extension_items = List().tag(sync=True)
+    extension_selected = Any().tag(sync=True)
+    extension_multiselect = Bool(True).tag(sync=True)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.app.config == 'imviz':
+            self.data_label_default = 'Image'
+
+        self.input_hdulist = isinstance(self.input, fits.HDUList)
+        if self.input_hdulist:
+            self.extension = SelectFileExtensionComponent(self,
+                                                          items='extension_items',
+                                                          selected='extension_selected',
+                                                          multiselect='extension_multiselect',
+                                                          manual_options=self.input,
+                                                          filters=[_validate_fits_image2d])
+            self.extension.selected = [self.extension.choices[0]]
+
+    @property
+    def is_valid(self):
+        if self.app.config not in ('deconfigged', 'imviz', 'cubeviz'):
+            # cubeviz allowed for cubeviz.specviz.load_data support
+            # NOTE: temporary during deconfig process
+            return False
+        # flat image, not a cube
+        # isinstance NDData
+        return isinstance(self.input, (fits.HDUList, NDData, np.ndarray, asdf.AsdfFile))
+
+
+    @property
+    def default_viewer_reference(self):
+        # returns the registry name of the default viewer
+        # only used if `show_in_viewer=True` and no existing viewers can accept the data
+        return 'imviz-0'
+
+    @property
+    def output(self):
+        # if nddata return it
+        if isinstance(self.input, NDData) or isinstance(self.input, np.ndarray):
+            return self.input
+        elif isinstance(self.input, asdf.AsdfFile) or (HAS_ROMAN_DATAMODELS and isinstance(self.input, rdd.DataModel)):
+            return self.input
+        hdulist = self.input
+        return [_hdu2data(hdu, self.data_label_value, hdulist)[0] for hdu in self.extension.selected_hdu]
+
+    def __call__(self):
+        data_label = self.data_label_value
+        if isinstance(self.output, NDData):
+            data = _nddata_to_glue_data(self.output, data_label)
+            self.add_to_data_collection(data, f"{data_label}[Array]", show_in_viewer=True)
+        elif isinstance(self.output, np.ndarray):
+            data = _ndarray_to_glue_data(self.output, data_label)
+            self.add_to_data_collection(data, f"{data_label}[Array]", show_in_viewer=True)
+        elif isinstance(self.output, asdf.AsdfFile) or (HAS_ROMAN_DATAMODELS and isinstance(self.output, rdd.DataModel)):
+            data = _roman_asdf_2d_to_glue_data(self.output, data_label)
+            self.add_to_data_collection(data, f"{data_label}[ASDF]", show_in_viewer=True)
```

Comment:
let's make a follow-up ticket to migrate these internal functions copied from the old parsers to glue-astronomy which would then really simplify this block to just handling the for-loop